save-cosv/src/main/kotlin/com/saveourtool/save/cosv/service/CosvUserService.kt

```diff
@@ -0,0 +1,85 @@
+package com.saveourtool.save.cosv.service
+
+import com.saveourtool.save.authservice.utils.userId
+import com.saveourtool.save.cosv.storage.AvatarStorage
+import com.saveourtool.save.domain.UserSaveStatus
+import com.saveourtool.save.entities.User
+import com.saveourtool.save.info.UserStatus
+import com.saveourtool.save.repository.LnkUserOrganizationRepository
+import com.saveourtool.save.repository.LnkUserProjectRepository
+import com.saveourtool.save.repository.OriginalLoginRepository
+import com.saveourtool.save.repository.UserRepository
+import com.saveourtool.save.service.UserService
+import com.saveourtool.save.storage.AvatarKey
+import com.saveourtool.save.utils.AvatarType
+import com.saveourtool.save.utils.orNotFound
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Primary
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.security.core.Authentication
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import reactor.core.scheduler.Schedulers
+
+/**
+ * Service for user
+ */
+@Service
+@Suppress("LongParameterList")
+@Primary
+class CosvUserService(
+    private val userRepository: UserRepository,
+    private val applicationEventPublisher: ApplicationEventPublisher,
+    private val originalLoginRepository: OriginalLoginRepository,
+    private val lnkUserOrganizationRepository: LnkUserOrganizationRepository,
+    private val lnkUserProjectRepository: LnkUserProjectRepository,
+    private val avatarStorage: AvatarStorage,
```

Comment:
change avatarStorage: configProperties: S3OperationsProperties.Provider,

Reply:
Changed it